82 countdown

psugihara opened the pull request:
pretty simple version of #82. I punted on a more complicated admin mode / toggle by just commenting old score stuff. Can add back if we want to do debugging outside dev environment but otherwise this saves a touch of bundle size.

Thoughts on placement or anything? Thinking about having it change color or something as it counts down.

<img width="894" alt="Screenshot 2024-04-01 at 3 45 35 PM" src="https://github.com/okwme/anybody-problem/assets/282016/232eb66f-1b06-42f8-9d34-47391d45af5e">

psugihara commented:
oh it could get really big and wild when it gets <10s 👅

okwme commented on src/visuals.js:
```diff
@@ -294,69 +298,106 @@ export const Visuals = {
+    p.textSize(this.scoreSize)
+
+    // TODO: remove this line when timer is ticked somewhere more suitable (okwme's branch)
+    this.timer--
```
hmm think this will break current timer logic. Could merit changing existing logic but right now during initialization `this.timer` value is combined with `this.alreadyRun` so that when `this.frames >= this.timer` it means time as run out.

would changing back logic of line 71 and removing this line break anything else in your branch currently?

psugihara replied:
i don't think it breaks anything as long as the timer ticks down somewhere as described in #82 

I first added this because timer was not moving, then changed that because the game was stopping at 30s (when timer and frames met)

okwme replied:
could we just use a new variable like `const timeLeft = this.timer - this.frames` ?

psugihara replied:
sure. timer counts up in your branch?

psugihara replied:
or like it is the frame at which the game ends?

okwme replied:
> or like it is the frame at which the game ends?
exactly

psugihara replied:
ok got it

psugihara commented:
refactored timer stuff as mentioned.

added quickie game over text because the alert was stopping animation

<img width="374" alt="Screenshot 2024-04-02 at 9 54 29 AM" src="https://github.com/okwme/anybody-problem/assets/282016/77da85b8-880d-406d-9515-88d0afffe23b">

at the end of countdown, game over shows and bodies all wither

okwme commented:
lmao love the double big up on the numbers and the GAMEOVER type face is very arcade 🔥
